src/Raven.Studio/typescript/models/database/index/additionalAssemblyModel.ts

```diff
@@ -0,0 +1,182 @@
+        this.assemblySource.subscribe((newSource) => {
+            this.usings([]);
+            
+            switch (newSource) {
+                case "Server Runtime":
+                    this.clearPathFields();
+                    this.clearNuGetFields();
+                    break;
+                case "Path":
+                    this.clearServerRuntimeFields();
+                    this.clearNuGetFields();
+                    break;
+                case "NuGet":
+                    this.clearServerRuntimeFields();
+                    this.clearPathFields();
+                    break;
+            }
+        })
+        
+        this.dirtyFlag = new ko.DirtyFlag([
+            this.assemblySource,
+            this.assemblyName,
+            this.assemblyPath,
+            this.packageName,
+            this.packageVersion,
+            this.packageSourceUrl,
+            this.usings
+        ], false, jsonUtil.newLineNormalizingHashFunction);
+    }
+    
+    private clearServerRuntimeFields() {
+        this.assemblyName(null);
+    }
+
+    private clearPathFields() {
+        this.assemblyPath(null);
+    }
+
+    private clearNuGetFields() {
+        this.packageName(null);
+        this.packageVersion(null);
+        this.packageSourceUrl(null);
+        this.useDefaultPackageSourceUrl(true);
+    }
+    
+    private initValidation() {
+        this.assemblyName.extend({
+            required: {
+                onlyIf: () => this.assemblySource() === "Server Runtime"
+            }
+        });
+
+        this.assemblyPath.extend({
+            required: {
+                onlyIf: () => this.assemblySource() === "Path"
+            }
+        });
+
+        this.packageName.extend({
+            required: {
+                onlyIf: () => this.assemblySource() === "NuGet"
+            }
+        });
+
+        this.packageVersion.extend({
+            required: {
+                onlyIf: () => this.assemblySource() === "NuGet"
+            }
+        });
+
+        this.validationGroup = ko.validatedObservable({
+            assemblyName: this.assemblyName,
+            assemblyPath: this.assemblyPath,
+            packageName: this.packageName,
+            packageVersion: this.packageVersion
+        });
+    }
+    
+    private computeAssemblySource(): assemblySource {
+        if (this.packageName()) {
+            return "NuGet";
+        }
+        
+        if (this.assemblyPath()) {
+            return "Path";
+        }
+        
+        return "Server Runtime";
+    }
+
+    setAssemblySourceType(sourceType: assemblySource) {
+        this.assemblySource(sourceType);
+    }
+
+    addNamespaceToUsingsWithBlink() {
+        const namespaceToAdd = this.namespaceText();
+
+        if (!this.usings().find(x => x === namespaceToAdd))
+        {
+            this.usings.unshift(namespaceToAdd);
+            this.namespaceText(null);
+            $(".usings .collection-list li").first().addClass("blink-style");
```

Comment:
can we move this logic to view? Like we talked 2 days ago: playing with UI directly in model is blending layers.

Reply:
done